### DISCUSSION TITLE
Break for end of sentence

### DISCUSSION THREAD

**Discussion by Cyprien-22** [SubtitleEdit/subtitleedit#7347] [Q&A]
One aspect that takes me the longest to edit subtitles is, often the first word of a new sentence is placed on the previous subtitle. 

![image](https://github.com/SubtitleEdit/subtitleedit/assets/144349105/db5d68c3-d1f6-47c4-ae31-c432284171e1)

I've tried "break early for end of sentence", but that simply put the single word on it's own line. 

![image](https://github.com/SubtitleEdit/subtitleedit/assets/144349105/e8f04a54-5df2-43c2-83b1-a8401c01d6c4)

I can't seem to find an option in "Fix common errors" 

Are there any options or methods I can quickly fix this? 

Can I generate a rule that's something like IF word count < 2 after a full stop, push to next subtitle? Sorry, I'm not a coder so not sure how it might be achievable.

**Comment by Purfview** [comment]
I would like this betterment too.
Splitting after "the" is annoying too.

**Comment by Cyprien-22** [comment]
A solution for both would be amazing! 

So much time saving right there. I just have absolutely no idea how to make it happen.

**Comment by Cyprien-22** [comment]
I do wonder, Fix common errors can already determine the end of a sentence, so the first part should not be difficult to implement?

But your suggestion of splitting after "the" might be a bit harder I imagine. Perhaps an alternative would be to "bump" highlighted text to the previous or next subtitle by way of a keyboard shortcut? 

My thinking is that the bumping text would be faster than cutting, pasting and re-timing the section. It would have to be limited to the start of the next subtitle, or end of the previous one. This way it's also more flexible for odd subtitle breaks to be quickly corrected in greater scope than just "the" breaks while being easier to potentially code?

Again, no idea how these would be implemented, sometimes simple thoughts are super hard to do through code. But if you don't share the thoughts, the possibility of implementation is zero.

**Comment by oep42** [comment]
> My thinking is that the bumping text would be faster than cutting, pasting and re-timing the section.

Cutting and pasting may not be necessary, when using one of the marked shortcuts in [the first image of this post](https://github.com/SubtitleEdit/subtitleedit/discussions/7344#discussioncomment-7146229).

In [another context](https://github.com/SubtitleEdit/subtitleedit/issues/6856), I suggested to automatically correct something similar, namely when the last word of a sentence is placed in the next subtitle. Perhaps it is difficult to automatically correct something like that.

By the way, automatically moving the first word of a sentence from the end of a subtitle line to the next subtitle line, may be problematic, if this is done for all the subtitles in a video, because you always have to check the corresponding timestamps for each of these moves and correct them if necessary. So if you do this as "Fix common errors" for an entire video, you won't know which timestamps to check because you won't be able to see where these moves have occurred. (Also, the second subtitle line of such a move, may become too long because of the move. Automatically splitting such a long subtitle line, may introduce new timestamp errors at the point of automatic splitting.)

In [yet another context](https://github.com/SubtitleEdit/subtitleedit/issues/7375), I suggested to allow editing directly in the list view, in such a way, that it will be possible to select a word (or the line break character), and then to drag it to another position in the same subtitle line. To expand on this, it would be cool if you could, in the list view, select a word in a subtitle line, and then drag it to the previous or next subtitle line. In this way, you could manually correct the beginning or end of a subtitle in an intuitive way (except for the timing), by just selecting the last word of a subtitle line and drag it to the beginning of the next subtitle line, or by selecting the first word of a subtitle line and drag it to the end of the previous subtitle line.

**Reply by HeikeMuc** [reply]
I wonder if it would be possible to write a regular expression into the "Do not break after" list. Many people have the problem that a line ends with a comma or a fullstop, and behind it one single word. So one could say "Fullstop, space, and behind it at least 1 or 2 letters, maximum 10 letters" or so. I am not familiar with regex but maybe someone knows.

**Comment by Purfview** [comment]
I implemented such there: https://github.com/Purfview/whisper-standalone-win/discussions/125

So, it's possible.

**Reply by HeikeMuc** [reply]
Wow, super! I am not good in programming. So do I have to copy these lines into the do not break after list?

**Reply by Purfview** [reply]
No, it works only there, in `whisper-faster.exe`.
